### PR TITLE
CRAYSAT-1649: add error logging when unable to find token file path

### PR DESCRIPTION
**Pull request by ethanholen-hpe**
## Summary and Scope

add error logging when unable to find token file path and unwrap ~ as home directory

## Issues and Related PRs

* Resolves [CRAYSAT-1649](https://jira-pro.it.hpe.com:8443/browse/CRAYSAT-1649)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge after https://github.com/Cray-HPE/python-csm-api-client/pull/48

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * Rocket
  * Drax
  * Local development environment

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by ethanholen-hpe** [comment]
https://gist.github.com/ethanholen-hpe/dfa1ad9b2575e5998ab7f0bd45dc5992

**Comment by ethanholen-hpe** [comment]
Testing on rocket: https://gist.github.com/ethanholen-hpe/f12537851c1c0b0cefb63d0142250195

File: requirements-dev.lock.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ coverage==6.3.2
 cray-product-catalog==2.4.1
 croniter==0.3.37
 cryptography==43.0.1
-csm-api-client==2.2.3
+csm-api-client==2.2.5
```
Comment (haasken-hpe) [inline requirements-dev.lock.txt:17]: Update to new version once you rebase the python-csm-api-client PR

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [3.32.13] - 2024-11-26
```
Comment (haasken-hpe) [inline CHANGELOG.md:28]: Needs to be rebased and bumped to 3.33

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,6 +25,12 @@ All notable changes to this project will be documented in this file.
...
+
+### Fixed
+-  Added error message when token file is not found, along with allowing for
```
Comment (haasken-hpe) [inline CHANGELOG.md:31]: Probably should add that it exits too

File: tests/cli/bootsys/test_service_activity.py [hunk 4]
```diff
@@ -401,6 +401,7 @@ class TestBOSV2ActivityChecker(unittest.TestCase):
 
     def setUp(self):
         self.mock_bos_client = MagicMock()
+        self.mock_sat_session = patch('sat.cli.bootsys.service_activity.SATSession').start()
```
Comment (haasken-hpe) [inline tests/cli/bootsys/test_service_activity.py:404]: I noticed that this `TestBOSV2ActivityChecker` class is missing a `tearDown` method that stops the patches. We should add that.

File: tests/cli/firmware/test_firmware.py [hunk 5]
```diff
@@ -134,7 +137,7 @@ def test_delete_snapshot(self):
         self.firmware_client.delete_snapshot.assert_called_once_with('snap1')
 
     def test_delete_snapshot_error(self):
-        """An APIError deleting incorrect snapshot name logs an error and exits"""
```
Comment (haasken-hpe) [inline tests/cli/firmware/test_firmware.py:137]: why was this deleted?

**Review by haasken-hpe — APPROVED** [review]
Looks good. Minor comments.

File: tests/cli/firmware/test_firmware.py (hunk 5 above)
Comment (ethanholen-hpe) [reply tests/cli/firmware/test_firmware.py:137]: not sure, just added it back 🤔
